maubot/server.py

```diff
@@ -103,7 +103,7 @@ def setup_management_ui(self) -> None:
             ui_base = ""
         directory = self.config[
             "server.override_resource_path"
-        ] or pkg_resources.resource_filename("maubot", "management/frontend/build")
+        ] or resources.files("maubot").joinpath("management/frontend/build")
```

Comment:
`directory` is a path, while `resources.files` returns an implementation of `importlib.resources.abc.Traversable`, which isn't guaranteed to be a path. This means that after you assign a `Traversable` to `directory`, you can't call `open(f"{directory}/foo", bar)` (as is done below), you must instead do `traversable.join("foo").open(bar)`.

Reply:
I'll go take a look into it, thanks for the review.

Reply:
I'm currently looking at it using Python 3.11 and 3.12. `resources.files(...).joinpath(...)` returns a `PosixPath`-Object. In what situation in this particular context would there be a risk of it not being a `pathlib.Path`-like object?

Reply:
This is *implementation details*. It's almost never okay to rely on implementation details.

Reply:
Right. My problem is that I'm not quite sure yet I understand the problem beyond the context of the implementation detail.

Reply:
[`importlib` documentation](https://docs.python.org/3/library/importlib.resources.html#importlib.resources.files) says that `files` returns an implementation of [`Traversable`](https://docs.python.org/3/library/importlib.resources.abc.html#importlib.resources.abc.Traversable). It's guaranteed to implement *a subset* of methods implemented by [`pathlib.Path`](https://docs.python.org/3/library/pathlib.html#pathlib.Path), but not guaranteed to *be* `pathlib.Path` despite the compatible interface.

Right now the code uses string paths, so it's not like `pathlib.Path` at all. If you change the existing code to use `pathlib.Path` instead before applying the `pkg_resources`->`importlib` changes, adding the `importlib` changes on top will be trivial.